frontend/src/components/viewers/ROCCurve.tsx

```diff
@@ -141,16 +146,39 @@ class ROCCurve extends Viewer<ROCCurveProps, ROCCurveState> {
           />
 
           {/* Lines */}
-          {datasets.map((data, i) => (
+          {datasets.map(
+            (data, i) =>
+              highlightIndex !== i && (
+                <LineSeries
+                  key={i}
+                  color={
+                    this.props.colors
+                      ? this.props.colors[i]
+                      : lineColors[i] || lineColors[lineColors.length - 1]
+                  }
+                  strokeWidth={2}
+                  data={data}
+                  onNearestX={(d: any) => this._lineHovered(i, d)}
+                  curve='curveBasis'
+                />
+              ),
+          )}
+
+          {/* Highlighted line, if present */}
+          {highlightIndex !== -1 && (
             <LineSeries
-              key={i}
-              color={lineColors[i] || lineColors[lineColors.length - 1]}
-              strokeWidth={2}
-              data={data}
-              onNearestX={(d: any) => this._lineHovered(i, d)}
+              key={highlightIndex}
+              color={
+                this.props.colors
+                  ? this.props.colors[highlightIndex]
+                  : lineColors[highlightIndex] || lineColors[lineColors.length - 1]
```

Comment:
Will it cause color conflict if we use `lineColors[lineColors.length - 1]`?

Reply:
This will cause a color conflict; however, this is the same code and logic from before, and the only fix would be to randomly generate new colors.

Further, the `lineColors[lineColors.length - 1]` line will not be reached on the ROC Curve metrics code, as the `this.props.colors` is passed in and the user is prevented from selecting more than `10` artifacts. Therefore, I decided to keep that `lineColors[lineColors.length - 1]` line as this issue should not present on the UI and the fix would be to generate random colors (which is a complex problem for a feature that would not be used).

Reply:
I see, sounds good on keeping it the current code then. Thank you for explanation!